### PR TITLE
add incident type validator

### PR DESCRIPTION
**Pull request by bakatzir**
## Status
Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/20084

### CODE REVIEW

File: demisto_sdk/commands/common/hook_validations/incident_type.py [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+        """
+        return super(IncidentTypeValidator, self)._is_valid_version()
+
+    def is_current_valid_from_version(self):
+        # type: () -> bool
+        """Check if the current file fromversion is valid.
+        Returns:
+            bool. Whether the current fromversion is valid or not.
+        """
+        error_msg = None
```
Comment (yuvalbenshalom) [inline demisto_sdk/commands/common/hook_validations/incident_type.py]: why do you need this var? isn't it better to just print any error that you find?

File: demisto_sdk/commands/common/hook_validations/incident_type.py [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+                is_bc_broke = True
+        return is_bc_broke
+
+    def is_including_int_fields(self):
+        # type: () -> bool
+        """Check if including required fields, only from 5.0.0.
+        Returns:
+            bool. Whether the fields .
+        """
+        error_msg = ""
```
Comment (yuvalbenshalom) [inline demisto_sdk/commands/common/hook_validations/incident_type.py]: here again, not sure if we need this var.

**Comment by bakatzir** [comment]
@yuvalbenshalom I can, but this is not the convention of the hook_validations that we have. and I find it readable.

**Comment by yuvalbenshalom** [comment]
@bakatzir I don't understand how it is more reliable to put an error you found in a variable, in hope that it will be printed later, rather than print it instantly. what are the pros for this?

**Comment by bakatzir** [comment]
@yuvalbenshalom Agreed.

**Pull request merged**